File: sort/insertion_sort/php/insertion_sort.php
<?php
    /**
     * @param Array  $array
     * 
     * returns a sorted array after applying insertion sort
     */
    function insertion_sort($array){
        for($i=1;$i<count($array); $i++){
            for($j=$i;$j >=1;$j--){
                if($array[$j] < $array[$j-1]){
                    $temp = $array[$j-1];
                    $array[$j-1] = $array[$j];
                    $array[$j] = $temp;
                } else{
                    break;
                }
            }
        }

        return $array;
    }

    //test
    function test(){
        $array = [19,138,1,6,5,41,2,2,12];
        $sorted = insertion_sort( $array);
        print_r($sorted);
    }

    test();

File: sort/insertion_sort/rust/insertion_sort.rs
fn insertion_sort(arr: &mut Vec<i32>) -> &Vec<i32> {
    let mut temp: &mut Vec<i32> = arr;
    for i in 0..temp.len(){
        let mut j = i;
        let mut swap;
        while (j > 0) && (temp[j - 1] > temp[j]){
            swap = temp[j];
            temp[j] = temp[j - 1];
            temp[j - 1] = swap;
            j = j - 1;
        }
    }
    return temp;
}

fn test() {
    let mut test: Vec<i32> = vec![22,84,67,1,90,15,88,23,69];
    let sorted_test = insertion_sort(&mut test);
    for i in sorted_test {
        println!("{}",*i);
    }
}

fn main() {
    test();
}
